api/tests/unit/integrations/lead_tracking/hubspot/test_unit_hubspot_lead_tracking.py

```diff
@@ -1,469 +1,348 @@
     responses.add(
-        method="POST",
-        url=url,
-        status=status.HTTP_200_OK,
-        json={"inlineMessage": "Thanks for submitting the form."},
-    )
-
-    HubspotTracker.objects.create(user=user, hubspot_cookie="tracker")
-    future_hubspot_id = "10280696017"
-    mock_create_company = mocker.patch(
-        "integrations.lead_tracking.hubspot.client.HubspotClient.create_company",
-        return_value={
-            "id": future_hubspot_id,
-            "properties": {
-                "createdate": "2024-02-26T19:41:57.959Z",
-                "hs_lastmodifieddate": "2024-02-26T19:41:57.959Z",
-                "hs_object_id": future_hubspot_id,
-                "hs_object_source": "INTEGRATION",
-                "hs_object_source_id": "2902325",
-                "hs_object_source_label": "INTEGRATION",
-                "name": organisation.name,
-            },
-            "properties_with_history": None,
-            "created_at": datetime.datetime(2024, 2, 26, 19, 41, 57, 959000),
-            "updated_at": datetime.datetime(2024, 2, 26, 19, 41, 57, 959000),
-            "archived": False,
-            "archived_at": None,
-        },
+        responses.POST,
+        f"{HUBSPOT_ROOT_FORM_URL}/{HUBSPOT_PORTAL_ID}/{HUBSPOT_FORM_ID}",
+        json={},
+        status=200,
+        match=[responses.matchers.header_matcher({"Content-Type": "application/json"})],
     )
-
-    mock_get_contact = mocker.patch(
+    mocker.patch(
         "integrations.lead_tracking.hubspot.client.HubspotClient.get_contact",
-        return_value=None,
-    )
-
-    hubspot_lead_id = "1000551"
-    mock_create_contact = mocker.patch(
-        "integrations.lead_tracking.hubspot.client.HubspotClient.create_contact",
-        return_value={
-            "archived": False,
-            "archived_at": None,
-            "created_at": datetime.datetime(2024, 2, 26, 20, 2, 50, 69000),
-            "id": hubspot_lead_id,
-            "properties": {
-                "createdate": "2024-02-26T20:02:50.069Z",
-                "email": user.email,
-                "firstname": user.first_name,
-                "hs_all_contact_vids": "1000551",
-                "hs_email_domain": "example.com",
-                "hs_is_contact": "true",
-                "hs_is_unworked": "true",
-                "hs_marketable_status": user.marketing_consent_given,
-                "hs_object_id": "1000551",
-                "hs_object_source": "INTEGRATION",
-                "hs_object_source_id": "2902325",
-                "hs_object_source_label": "INTEGRATION",
-                "hs_pipeline": "contacts-lifecycle-pipeline",
-                "lastmodifieddate": "2024-02-26T20:02:50.069Z",
-                "lastname": user.last_name,
-            },
-            "properties_with_history": None,
-            "updated_at": datetime.datetime(2024, 2, 26, 20, 2, 50, 69000),
-        },
+        return_value={"id": HUBSPOT_USER_ID},
```

Comment:
We should perhaps add a comment here explaining that we're mocking the response to get_contact to simulate the situation where hubspot creates the contact immediately after we submit the form?